unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfileModel.cs

```diff
@@ -93,15 +93,19 @@ public bool Equals(UserProfileModel model)
 
         if (model.blocked != null && blocked != null)
         {
-            if (!model.blocked.All(blocked.Contains) || model.blocked.Count != blocked.Count)
+            if (model.blocked.Count != blocked.Count ||
```

Comment:
i think we could gain more performance by using a `HashSet` for the `blocked` & `muted` lists, what do you think?

Reply:
I have tried but using directly `HashSet` as part of the `UserProfileModel` model, it cannot serialize well the `blocked` and `muted` lists that comes from kernel.

I could try to convert the lists to `HashSet` just before doing the comparisions, but I'm not sure if it would worth it since we would have to duplicate the data. Wdyt?

Reply:
We've discussed to remove the content comparison of both lists and only do the size checking. We need a proper fix in the backend. The lists being too big could cause hiccups, not only here, but in many places when processing the profile